partner_time_to_pay/tests/test_partner_time_to_pay.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+from odoo.tests.common import TransactionCase
+from dateutil.relativedelta import relativedelta
+
+
+class TestPartnerTimeToPay(TransactionCase):
+
+    def setUp(self):
+        super(TestPartnerTimeToPay, self).setUp()
+        self.payment_model = self.env['account.payment']
+        self.register_payments_model = self.env['account.register.payments']
+        self.partner_id = self.env.ref('base.res_partner_1')
+        self.product_id = self.env.ref('product.product_product_4')
+        self.account_model = self.env['account.account']
+        self.account_invoice_model = self.env['account.invoice']
+        self.account_journal = self.env['account.journal']
+        self.payment_method_manual_in =\
+            self.env.ref("account.account_payment_method_manual_in")
+        self.account_receivable_id =\
+            self.env.ref('account.data_account_type_receivable')
+        self.account_revenue_id =\
+            self.env.ref('account.data_account_type_revenue')
+
+        self.account_receivable = self.account_model.search(
+            [('user_type_id', '=', self.account_receivable_id.id)], limit=1)
+        self.account_revenue = self.account_model.search(
+            [('user_type_id', '=', self.account_revenue_id.id)], limit=1)
+
+        self.invoice_id = self.account_invoice_model.create({
+            'partner_id': self.partner_id.id,
+            'type': 'out_invoice',
+            'account_id': self.account_receivable.id,
+            'date_invoice': datetime.datetime.now().date(),
```

Comment:
You can store today in a new variable and then use it here and a bit further down as well.
```
from odoo.fields import Date
self.today = Date.from_string(Date.context_today(self.env.user))
```

Reply:
@PieterPaulussen It's Not needed, `datetime.datetime.now().date()` It's get current date too.

Reply:
I disagree. You can have date time differences due to timezones of users. Please use Date.context_today()...